### PR TITLE
add trustServerCertificate prop to MSSQL driver extension

### PR DESCRIPTION
**Pull request by HasseNasse**
This issue aims to resolve the need to manually add the property:
`quarkus.datasource.jdbs.additional-jdbc-properties.trustservercertificate=true`

When using the quarkus-jdbc-mssql, the end-user is often not aware of the underlying mssql driver specifics, and expects the vendor/implementor to handle the specifics. In our organization we experienced issues when moving from Quarkus 2.9.2 --> 2.14.2 that suddenly our MSSQL connections were failing with a PKIX path building error.

```
Datasource '<default>': The driver could not establish a secure connection to SQL Server by using Secure Sockets Layer (SSL) encryption. Error: "PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target". ClientConnectionId:f2088dcd-2276-40b0-b7e4-90aac3a88eef
com.microsoft.sqlserver.jdbc.SQLServerException: The driver could not establish a secure connection to SQL Server by using Secure Sockets Layer (SSL) encryption. Error: "PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target". ClientConnectionId:f2088dcd-2276-40b0-b7e4-90aac3a88eef
        at com.microsoft.sqlserver.jdbc.SQLServerConnection.terminate(SQLServerConnection.java:3806)
        at com.microsoft.sqlserver.jdbc.TDSChannel.enableSSL(IOBuffer.java:1906)
        at com.microsoft.sqlserver.jdbc.SQLServerConnection.connectHelper(SQLServerConnection.java:3329)
        at com.microsoft.sqlserver.jdbc.SQLServerConnection.login(SQLServerConnection.java:2950)
        at com.microsoft.sqlserver.jdbc.SQLServerConnection.connectInternal(SQLServerConnection.java:2790)
        at com.microsoft.sqlserver.jdbc.SQLServerConnection.connect(SQLServerConnection.java:1663)
        at com.microsoft.sqlserver.jdbc.SQLServerDriver.connect(SQLServerDriver.java:1064)
        at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:226)
        at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:535)
        at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:516)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: javax.net.ssl.SSLHandshakeException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
```

I may be in the wrong, but I assume that whenever we are using the MSSQL JDBC driver in tandemn with the quarkus-jdbc-mssql extension, it should default to using the server certificates for TLS hand-shakes. Whereas edge-cases should be handled as.. well.. edge cases. I.e when an override is required (maybe due to requiring a speicific keystore/certificate) then it's natural to expect the developer having to resort to the driver specific docs.

Please let me know if you agree @Sanne.

Connected to the issue here:
https://github.com/quarkusio/quarkus/issues/27330

### CODE REVIEW

**Comment by sberyozkin** [comment]
@HasseNasse  Can you clarify please why is this update done in the `disableSsl` method, as the problem above is related to establishing an SSL connection ?

**Comment by gsmet** [comment]
Yeah the idea of `disableSslSupport` is to not use SSL at all as it wouldn't work in native (at least it wasn't working back in the time without explicit configuration, I'm not sure it is still true).

**Comment by HasseNasse** [comment]
Yes I realize I simply put it in the first place w/o really understanding where configs should be set.
I've skimmed through the extension and can't really find where to add the following configuration property. I'm very new to extensions so would appreciate to get some pointers. 

Am I supposed to register a `@Recorder(RUNTIME_INIT)` and then inject the configurer `AgroalConnectionConfigurer` to add the configuration property?

I'll test it out thoroughly against our DB environments SSL afterwards.

**Comment by gsmet** [comment]
You cannot disable the verification of certificates by default because it would become a security risk.

So you will always have to set a configuration property for it.

**Review by Sanne — CHANGES_REQUESTED** [review]
I agree with @gsmet , we can't disable this by default. I also agree this is inconvenient, but that's how it is - perhaps you should complain to the microsoft sql developers about how this is structured, but I'm not sure if they can improve.

Perhaps we could improve a couple things though?

some random ideas could be:

- expose a regular configuration property for this so to not need to use the "additional properties"; might help making it a bit more convenient with auto-completion support from IDEs.
- perhaps we can find a way to trust the Testctonainers Image used for devservices? I'm not 100% if that's a good idea and I'd need to ask a second opinion from the security team but that's a possibility - what do you think?
- others? Feel free to suggest :)

Sorry for being picky about this, I do agree we need to improve it in some way and admit I couldn't spend time on this - please don't give up :)

Thanks!

**Comment by gsmet** [comment]
Closing. Thanks for the effort anyway!

**Pull request closed without merging**
